### PR TITLE
chore(golang): fix lint errors and enable lint errors

### LINKED ISSUE #4791: err in swingset/types/types.go isn't checked ineffassign
## Describe the bug
    https://github.com/Agoric/agoric-sdk/blob/574c026425425246aecad1b75e8d420cfb7c7806/golang/cosmos/x/swingset/types/types.go#L90

the `err` from Nat assigned, but not checked during the unmarshal method:
```
            ret.Nums[i], err = Nat(numFloat)
            msg, ok := nummsg[1].(string)
```

err should be checked and handled

### LINKED ISSUE #4793: Error return value of `am.keeper.SendCoinsToFeeCollector` is not checked (errcheck)
## Describe the bug
```
x/vbank/module.go:199:36: Error return value of `am.keeper.SendCoinsToFeeCollector` is not checked (errcheck)
		am.keeper.SendCoinsToFeeCollector(ctx, xfer)
```

Error should be collected and handled from `SendCoinsToFeeCollector`

https://github.com/Agoric/agoric-sdk/blob/adfdf3638f56e8f37d34661f033f42843354db26/golang/cosmos/x/vbank/module.go#L198-L202

### LINKED ISSUE #4794: Error return value of `am.PushAction` is not checked (errcheck)
## Describe the bug
```
x/vbank/module.go:192:16: Error return value of `am.PushAction` is not checked (errcheck)
		am.PushAction(ctx, action)	             ^
```
https://github.com/Agoric/agoric-sdk/blob/adfdf3638f56e8f37d34661f033f42843354db26/golang/cosmos/x/vbank/module.go#L191-L193

the error from `am.PushAction` is not handled currently

### PR DESCRIPTION
**Pull request by arirubinstein**
closes: #4791 
closes: #4794
closes: #4793

## Description

Adds error handling to the three cases discovered during restival, and enables the github linting workflow to error

### CODE REVIEW

File: golang/cosmos/x/vbank/module.go [hunk 1]
```diff
@@ -188,14 +188,21 @@ func (am AppModule) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.V
 	// Dump all the addressToBalances entries to SwingSet.
 	action := getBalanceUpdate(ctx, am.keeper, addressToBalance)
 	if action != nil {
-		am.PushAction(ctx, action)
+		err := am.PushAction(ctx, action)
+		if err != nil {
+			panic(err)
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/module.go:193]: I think it's better to log here and continue, as we do at 181 and 204, but I'm going to invite @michaelfig for an opinion. Giving an LGTM in advance of whatever decision is made.

**Review by JimLarson — APPROVED** [review]
Approved pending change to `vbank/module.go` after discussion with mfig.

File: golang/cosmos/x/vbank/module.go (hunk 1 above)
Comment (arirubinstein) [reply golang/cosmos/x/vbank/module.go:193]: Yikes, I had automerge on this. will follow up oob

Comment (michaelfig) [reply golang/cosmos/x/vbank/module.go:193]: It's fine with me.

**Pull request merged**